roles/rsyslog/defaults/main.yaml

```diff
@@ -67,10 +67,10 @@ rsyslog_base_packages: ['rsyslog']
 # List of rpm packages required for TLS support.
 rsyslog_tls_packages: ['rsyslog-gnutls', 'ca-certificates']
 
-# rsyslog_packages
+# rsyslog_additional_packages
 #
 # List of additional rpm packages to install.
-rsyslog_packages: []
+rsyslog_additional_packages: []
```

Comment:
Do you use this anywhere? IMO this only complicates the role and breaks abstraction. If installing extra packages is needed for some functionality, there should be a parameter for the functionality and the role should know what packages to install, instead of requiring the user to know this (it may be different for different supported distributions and versions).
If we decide to keep this functionality, I suggest to follow the example of OASIS, where it is suggested to be named extra_packages instead of additional_packages: https://github.com/oasis-roles/meta_standards#vars-vs-defaults
@robled @greg-hellings @seandst have you found such a variable useful in practice?

Reply:
I understand the issue, but I reluctant to remove this variable...
The rsyslog configuration is quite flexible and it supports these plugins.  Some of them are not installed by the current logging role.  But customers may want to manually configure it in their environment.  To cover such a use case, this rsyslog_additional_packages might be useful.
```
rsyslog-crypto-8.1911.0-3.el8.x86_64.rpm
rsyslog-elasticsearch-8.1911.0-3.el8.x86_64.rpm
rsyslog-gnutls-8.1911.0-3.el8.x86_64.rpm
rsyslog-gssapi-8.1911.0-3.el8.x86_64.rpm
rsyslog-kafka-8.1911.0-3.el8.x86_64.rpm
rsyslog-mmaudit-8.1911.0-3.el8.x86_64.rpm
rsyslog-mmjsonparse-8.1911.0-3.el8.x86_64.rpm
rsyslog-mmkubernetes-8.1911.0-3.el8.x86_64.rpm
rsyslog-mmnormalize-8.1911.0-3.el8.x86_64.rpm
rsyslog-mmsnmptrapd-8.1911.0-3.el8.x86_64.rpm
rsyslog-mysql-8.1911.0-3.el8.x86_64.rpm
rsyslog-pgsql-8.1911.0-3.el8.x86_64.rpm
rsyslog-relp-8.1911.0-3.el8.x86_64.rpm
rsyslog-snmp-8.1911.0-3.el8.x86_64.rpm
```

Reply:
I see. Still, can't you introduce more generic variables to control whether a specific functionality will be enabled and the plugin installed? I have tho concerns:
- the name of a plugin implementing a specific functionality can change, the role should isolate the user from such changes, but if the user has to know the exact names of packages, it does not fulfil this function,
- presumably to make use of a plugin, sometimes it could be necessary to configure it in addition to installing it (is that the case?), and this again calls for a more high-level approach, because when the role just installs given plugins, it will not help the user with configuring them.
If you decide to keep the variable, I would suggest to at least rename it to the OASIS convention.

Reply:
I renamed the misleading var name rsyslog_additional_packages with rsyslog_extra_packages as suggested.